perfkitbenchmarker/linux_benchmarks/cloudsuite_websearch_benchmark.py

```diff
@@ -211,120 +143,60 @@ def Run(benchmark_spec):
-                              '</fa:rampDown>" deploy/run.xml && '
-                              'sed -i "/<fa:steadyState>/c\<fa:steadyState>{6}'
-                              '</fa:steadyState>" deploy/run.xml '.format(
-                                  faban.FABAN_HOME_DIR,
-                                  benchmark_spec.vms[0].ip_address,
-                                  FABAN_OUTPUT_DIR, SOLR_PORT, terms_file,
-                                  FLAGS.cs_websearch_ramp_up,
-                                  FLAGS.cs_websearch_steady_state,
-                                  FLAGS.cs_websearch_scale, agents))
-  agent_id = 1
-  client_heap_size = FLAGS.cs_websearch_client_heap_size
-  driver_dir = posixpath.join(faban.FABAN_HOME_DIR, 'search')
-  policy_path = posixpath.join(driver_dir, 'config/security/driver.policy')
-  for vm in clients:
-    if vm == client_master:
-      faban.StartRegistry(vm, CLASSPATH, policy_path)
-      faban.StartAgent(vm, CLASSPATH, driver_dir, 'SearchDriver', agent_id,
-                       client_heap_size, policy_path, client_master.ip_address)
-    else:
-      faban.StartAgent(vm, CLASSPATH, driver_dir, 'SearchDriver', agent_id,
-                       client_heap_size, policy_path, client_master.ip_address)
-    agent_id = agent_id + 1
-  benchmark_config = posixpath.join(faban.FABAN_HOME_DIR,
-                                    'search/deploy/run.xml')
-  faban.StartMaster(client_master, CLASSPATH, client_heap_size,
-                    policy_path, benchmark_config)
-
-  def ParseOutput(client):
-    stdout, _ = client.RemoteCommand('cd {0} && '
-                                     'cd `ls -Art | tail -n 1` && '
-                                     'cat summary.xml'.format(
-                                         FABAN_OUTPUT_DIR))
-    ops_per_sec = re.findall(r'\<metric unit="ops/sec"\>(\d+\.?\d*)', stdout)
-    sum_ops_per_sec = 0.0
-    for value in ops_per_sec:
-        sum_ops_per_sec += float(value)
-    p90 = re.findall(r'\<p90th\>(\d+\.?\d*)', stdout)
-    sum_p90 = 0.0
-    for value in p90:
-        sum_p90 += float(value)
-    p99 = re.findall(r'\<p99th\>(\d+\.?\d*)', stdout)
-    sum_p99 = 0.0
-    for value in p99:
-        sum_p99 += float(value)
-    return sum_ops_per_sec, sum_p90, sum_p99
+  clients = benchmark_spec.vm_groups['clients'][0]
+  servers = benchmark_spec.vm_groups['servers'][0]
+  results = []
+
+  benchmark_cmd = ('sudo docker run --rm --net host --name client '
+                   'cloudsuite/web-search:client %s %d %d %d %d ' %
+                   (servers.internal_ip,
+                    FLAGS.cs_websearch_scale,
+                    FLAGS.cs_websearch_ramp_up,
+                    FLAGS.cs_websearch_steady_state,
+                    FLAGS.cs_websearch_ramp_down))
+  stdout, _ = clients.RemoteCommand(benchmark_cmd, should_log=True)
+
+ # time.sleep(3600)
+
+  ops_per_sec = re.findall(r'\<metric unit="ops/sec"\>(\d+\.?\d*)', stdout)
```

Comment:
This coding pattern of findall and calculating a sum occurs three times. Consider a helper function. Also consider verifying that the number of values found matches your expectation (should it equal the number of clients?)